lib/grpc/adapter/client.ex

```diff
@@ -0,0 +1,27 @@
+defmodule GRPC.ClientAdapter do
```

Comment:
```suggestion
defmodule GRPC.Client do
```

I think `ClientAdapter` is kinda redundant since the "adapter" part is actually what implements the behaviour specified here

Reply:
I think `GRPC.Ports.Client/Server` might be a better naming pattern (from the Ports and Adapters design pattern), and then we could have `GRPC.Adapters.Client.Gun` or something like that

Reply:
Just my opinion, curently we have `lib/grpc/client` and `lib/grpc/server` which provides client code and server code separately. How about we move the adapter behaviour to inside those folders? Like:

```
# For server adapter behaviour.
lib/grpc/server/adapter.ex
# For client adapter behaviour.
lib/grpc/client/adapter.ex
```

Reply:
I tend to prefer grouping all behaviours under a given namespace to make it easy to find all of them (so adapters.xxx), but I'm not set too hard on either option :)

Reply:
I’m ok with either option. :)

BTW It may apply to interceptor code to make it to the same pattern.

Reply:
I was thinking about grouping the code by context, something like this:

- /client/
  - stream.ex
  - adapter.ex
  - adapter/
    - gun.ex
- /server.ex
- /server/
  - stream.ex
  - supervisor.ex
  - adapter.ex
  - adapter/
    - cowboy.ex
    - cowboy/
      - handler.ex

WDYT? You guys have a bigger picture and will work on the project more than me 😅 , so let me know which one suits better to you 😄

BTW, @polvalente if you are in a rush to merge this, please, go ahead and take the PR if you want!! In the end, I can only take this from time to time...

Reply:
I agree with the suggested tree, but I would have the `adapters` directory instead of `adapter`

Reply:
> I agree with the suggested tree, but I would have the `adapters` directory instead of `adapter`

Nice! Agree with that! I'll push a commit with that ASAP 😄